warehouse/oidc/views.py

```diff
@@ -191,7 +191,10 @@ def _invalid(errors):
             f"({datetime.fromtimestamp(not_before).isoformat()})"
         ),
         [
-            caveats.OIDCPublisher(oidc_publisher_id=str(publisher.id)),
+            caveats.OIDCPublisher(
+                oidc_publisher_id=str(publisher.id),
+                oidc_claims={"ref": claims["ref"], "sha": claims["sha"]},
```

Comment:
One thing to note: `ref` should always be present, but the GitHub docs are a little bit slippery on whether `sha` always is.

For the time being, it might be worth making these fallible and treating them as progressive enhancements in the subsequent PR:

```suggestion
                oidc_claims={"ref": claims.get("ref"), "sha": claims.get("sha")},
```

Reply:
(And correspondingly, let's add tests for when they're missing.)